web-app/js/campaigns/contest.js

```diff
@@ -21,10 +21,22 @@ $(function () {
         e.preventDefault();
         var $a = $(this);
         console.log($a)
-        $a.parents("ul.nav").find("li").removeClass("active")
-        $a.parent().addClass("active")
-        var listSelector = $a.attr("data-listSelector");
-        contestApplicationHelper.showListApplications(listSelector);
+        if(!$a.parent().hasClass("active")) {
+            $a.parents("ul.nav").find("li").removeClass("active")
+            $a.parent().addClass("active")
+            var listSelector = $a.attr("data-listSelector");
+
+            //If sorted by number of votes is selected, data is refreshed.
+            if(listSelector === "vote") {
+                var $ul = $("#" + contestApplicationHelper.containerId).find(".search-list." + listSelector);
```

Comment:
Put this block of actions on a method. Something like clearList(selector)

Reply:
Putted into new function called setActiveAndRefreshList.